libnopegl/src/node_text.c

```diff
@@ -325,6 +325,9 @@ static int update_character_geometries(struct ngl_node *node)
             ngli_pipeline_compat_update_attribute(desc->pipeline_compat, 3, s->transforms);
 
             ngli_pipeline_compat_update_attribute(desc->pipeline_compat, 4, s->atlas_coords);
+
+            if (s->text_ctx->cls->mutable_texture)
+                ngli_pipeline_compat_update_texture(desc->pipeline_compat, 0, s->text_ctx->texture);
```

Comment:
I think you can do that unconditionally, update_character_geometries() is only called on a live change.

Reply:
Ok, so, isn't there a more appropriate mechanism to signal that the underlying texture has been updated (like a bool in the text ctx, `s->text_ctx->texture_has_changed` or a callback ) ? I believe the atlas won't be recreated every frame, right ?

Reply:
The incoming external font support won't have that much live change events so it's not problematic. Only the builtin one needs to be fast.

Any text change implies an atlas recreation in the case of external font because we are building an atlas per Text instance, which contains only the glyph used in the specified string (otherwise the atlas would be gigantic with certain fonts). If we were to rely on live change update spamming with external fonts, the texture update performance issue addressed here would be the least of our trouble (the re-rasterization of the atlas is going to be much more expensive).

Reply:
Would it make sense to call the field flags and have ..._FLAG_ATLAS_MUTABLE_BIT ? so it is more extensive.

Reply:
Sure, changed.